### ISSUE TITLE
Accept Date instances in network helpers

### ISSUE THREAD

**Issue by fvictorio** [NomicFoundation/hardhat#3530]
We have a bunch of helpers that accept timestamps, but they don't accept `Date` objects.

**Comment by Saty248** [comment]
@fvictorio i would like to work on this issue, 
as u said a bunch of helpers that accept timestamps, for that did you mean the helpers which are in the directory /packages/hardhat-network-helpers/src/helpers/time

**Comment by fvictorio** [comment]
Yes, some of them. I think the two ones that accept timestamps are `increaseTo` and `setNextBlockTimestamp`. You can see the full list [here](https://hardhat.org/hardhat-network-helpers/docs/reference).

**Comment by fvictorio** [comment]
Actually, `setNextBlockTimestamp` already accepts `Date`s, so apparently it's just `increaseTo`.
